### ISSUE TITLE
OCR the whole Window

### ISSUE THREAD

**Issue by Molitvan** [nvaccess/nvda#16282]
### Is your feature request related to a problem? Please describe.
There are a few problems I've had that would be solved by adding this. For example, there are a lot of situations where an inaccessible app will have a few things NVDA can focus on, which means it takes the focus away from the window and it is not possible to OCR it. Also, sometimes when I try to OCR something, it just says "Content not visible". Another thing is that I use OCR to play games, and sometimes it just glitches out. One very good example is Minecraft, where if you make it fullscreen and OCR it, it will recognize the last thing it saw before you made it fullscreen no matter what.
### Describe the solution you'd like
Make it possible to OCR the whole window or even the whole screen by pressing a hotkey.
### Describe alternatives you've considered
I don't know of any way to do this currently, even with add-ons.
### Additional context
N/A

**Comment by XLTechie** [comment]
Which add-ons have you tried?

CC @ruifontes

**Comment by ruifontes** [comment]
@XLTechie, our add-on Tesseract do not make OCR to screen contents...

**Comment by Brian1Gaff** [comment]
Its fine if its a Jpg of course, although it has no intelligence about columns etc, since there is no way to let it know about formatting.
Brian

-- 
***@***.*** 
Sent via blueyonder.(Virgin media) 
Please address personal E-mail to:-
***@***.***, putting 'Brian Gaff'
in the display name field.
  ----- Original Message -----

**Comment by lukaszgo1** [comment]
It is important to point out that `NVDA+R` OCR's the current navigator object. Therefore you can just move it either to the application window, or if you want to recognize the entire screen up all the way to the desktop. Will this be sufficient for your needs?

**Comment by Molitvan** [comment]
Thank you for responding. I've tried basically every add-on that has OCR in it's name or description. About focusing the desktop and then using OCR, it works sometimes, but when you have multiple apps opened at the same time, it can be hard to recognize which app the text is coming from. Also, when an app is fullscreen, you can't exactly focus the desktop.

**Comment by lukaszgo1** [comment]
It seems you have misunderstood me. I didn't suggest you should move to the desktop. Are you familiar with the object navigation? If you aren't I suggest to look it up in the user guide, and then try navigating with it to the desktop window without changing your current focus position.

**Comment by cary-rowen** [comment]
Hi,

Please note that I don't think this should be an issue, we really don't need to add this feature.
As @lukaszgo1 said:
We can completely use NVDA's object navigation to set the current navigator to any visible window.

In original description you said:
> Make it possible to OCR the whole window or even the whole screen by pressing a hotkey.

Full-screen OCR does not solve the problem you mentioned, namely: "but when you have multiple apps opened at the same time, it can be hard to recognize which app the text is coming from."


Have to ask again, are you familiar with object navigation? Could you first understand object navigation and then describe the problem based on the limitations of object navigation?


Thanks

**Comment by Molitvan** [comment]
Closing as object navigation does mostly solve my problem. The only thing that remains an issue is that weird behavior when OCR-ing Minecraft in full screen, but at this point I think it probably has to do with something the game itself does that NVDA can't do anything about.
